### ISSUE TITLE
[ListItem] ListItem is not forwarding refs

### ISSUE THREAD

**Issue by mkaradeniz** [chakra-ui/chakra-ui#1518]
**Describe the bug**

The `<ListItem />` component is not forwardings `ref`s.

**Expected Behavior** 

The `<ListItem />` component *is* forwardings `ref`s.

**Steps to Reproduce** 

See the code: https://github.com/chakra-ui/chakra-ui/blob/develop/packages/layout/src/list.tsx#L97

The `<ListItem />` is defined as:

```tsx
export const ListItem = (props: PropsOf<typeof chakra.li>) => (
  <chakra.li {...props} />
)
```

**Suggested solution(s)**
```tsx
export const ListItem = React.forwardRef(function ListItem(
  props: PropsOf<typeof chakra.li>,
  ref: React.Ref<any>,
) {
  return (
    <chakra.li ref={ref} {...props} />
  )
})
```

I think this would be it? But I might be missing something and there is a reason for this. For a workaround I used `<Box as="li" />`.

**Comment by ljosberinn** [comment]
Thank you, I'll take care of that when merging https://github.com/chakra-ui/chakra-ui/issues/1518 later today!

**Comment by ljosberinn** [comment]
naturally, I forgot it, but I'll add it as soon as Sage is done with his `forwardRef` change.

**Comment by segunadebayo** [comment]
I managed to fix it before seeing this issue. Thanks for catching that @mkaradeniz.

**Comment by silviuaavram** [comment]
Hi! Any updates on this one? I don't see the code changes in master yet.

**Comment by with-heart** [comment]
@silviuaavram I'm not sure what happened with the previous fix, but I opened #2794 to resolve this issue (again, apparently 😆).
